src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -41,6 +41,16 @@ public static bool CanBeConst(this TypeSymbol typeSymbol)
             return typeSymbol.IsReferenceType || typeSymbol.IsEnumType() || typeSymbol.SpecialType.CanBeConst();
         }
 
+        public static bool IsUnconstrainedTypeParameter(this TypeSymbol type)
+        {
+            if (type.TypeKind != TypeKind.TypeParameter)
+            {
+                return false;
+            }
+            var typeParameter = (TypeParameterSymbol)type;
+            return !typeParameter.IsValueType && !typeParameter.IsReferenceType;
```

Comment:
If I understood correctly from discussion on `unmanaged`, some other constraints imply a `struct` or a `class` constraint. I assume the `IsValueType` and `IsReferenceType` properly factor in such logic.
Maybe add a test for such a scenario?

Reply:
This branch does not include `unmanaged` support yet. Added `// PROTOTYPE` comment to test that case in the future.

---
In reply to: [177567274](https://github.com/dotnet/roslyn/pull/25733#discussion_r177567274) [](ancestors = 177567274)